nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/utils/AzureStorageUtils.java

```diff
@@ -143,6 +144,54 @@
             .required(false)
             .build();
 
+    public static final PropertyDescriptor ACCOUNT_NAME_SECRET = new PropertyDescriptor.Builder()
+            .name("storage-account-name-secret")
+            .displayName("Storage Account Name Secret Name")
+            .description("Storage Account Name Secret Name")
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .required(true)
+            .sensitive(true)
+            .build();
+
+    public static final PropertyDescriptor ACCOUNT_KEY_SECRET = new PropertyDescriptor.Builder()
+            .name("storage-account-key-secret")
+            .displayName("Storage Account Key Secret Name")
+            .description("Storage Account Key Secret Name")
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .required(false)
+            .sensitive(true)
+            .build();
+
+    public static final PropertyDescriptor ACCOUNT_SAS_TOKEN_SECRET = new PropertyDescriptor.Builder()
+            .name("storage-sas-token")
+            .displayName("SAS Token Secret Name")
+            .description("SAS Token Secret Name")
+            .required(false)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .sensitive(true)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor KEYVAULT_CONNECTION_SERVICE = new PropertyDescriptor.Builder()
+            .name("azure-keyvault-connection-service")
+            .displayName("KeyVault Connection Service")
+            .description("If configured, the controller service used to obtain the secrets from KeyVault")
+            .required(true)
+            .identifiesControllerService(AzureKeyVaultConnectionService.class)
+            .build();
+
+    public static final PropertyDescriptor ADLS_ENDPOINT_SUFFIX = new PropertyDescriptor.Builder()
+            .fromPropertyDescriptor(AzureStorageUtils.ENDPOINT_SUFFIX)
+            .displayName("Endpoint Suffix")
+            .description(
+                "Storage accounts in public Azure always use a common FQDN suffix. " +
+                    "Override this endpoint suffix with a different suffix in certain circumstances (like Azure Stack or non-public Azure regions).")
+            .required(true)
+            .defaultValue("dfs.core.windows.net")
+            .build();
```

Comment:
Should this property descriptor have some kind of validator, such as the NON_BLANK_VALIDATOR, or is blank valid?

Reply:
it is ok to be blank, default is blank.